compute/api/Compute.Samples/ListZoneInstancesAsync.cs

```diff
@@ -31,30 +31,15 @@ public class ListZoneInstancesAsyncSample
         // Initialize client that will be used to send requests. This client only needs to be created
         // once, and can be reused for multiple requests.
         InstancesClient client = await InstancesClient.CreateAsync();
-        InstanceList instanceList;
         IList<Instance> allInstances = new List<Instance>();
 
         // Make the request to list all VM instances in the given zone in the specified project.
-        ListInstancesRequest request = new ListInstancesRequest 
-        { 
-            Project = projectId,
-            Zone = zone,
-        };
-        do
+        await foreach(var instance in client.ListAsync(projectId, zone))
```

Comment:
An alternative to await foreach here would be to `await client.ListAsync(projectId, zone).ToListAsync()` to come up with `allInstances`, then synchronously iterate over that, but I'm not bothered either way.

Reply:
This one gives you he chance to break in the middle of the async iteration without having fetched everything, right? Even if we are not showing that,...

Reply:
Yes, it does. (And will show results as it goes, of course.) I just twitch at manually adding to the list ;)

Reply:
Ah yes, that's for the purposes of the tests only, but I do see what you mean about that making you twitch. We have several tests that do this kind of thing across the repo. It's the alternative to testing against stdout.